### PR TITLE
Added capability to generate a 6D Kurth beam distribution.

### PR DESCRIPTION
**Pull request by cemitch99**
This can be used later for benchmarks involving 3D space charge.

### CODE REVIEW

File: src/ImpactX.cpp [hunk 1]
```diff
@@ -353,8 +354,62 @@ namespace impactx
           m_particle_container->AddNParticles(lev, x, y, t, px, py, pt,
                                               qm, bunch_charge);
 
+        } else if (distribution_type == "kurth6d") {
```
Comment (ax3l) [inline src/ImpactX.cpp:357]: Awesome!
Please add in `source/usage/parameters.rst` as well :)

File: src/particles/distribution/Kurth6D.H [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+namespace distribution
+{
+    struct Kurth6D
+    {
+        /** A 6D Kurth distribution
+         *
+         * Return sampling from a 6D Kurth distribution.
+         *
+         * @param sigx,sigy,sigt for zero correlation, these are the related
+         *                                   RMS sizes (length); ...
```
Comment (ax3l) [inline src/particles/distribution/Kurth6D.H]: leftover `...`?

File: src/particles/distribution/Kurth6D.H [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+
+namespace impactx
+{
+namespace distribution
+{
+    struct Kurth6D
+    {
+        /** A 6D Kurth distribution
+         *
+         * Return sampling from a 6D Kurth distribution.
```
Comment (ax3l) [inline src/particles/distribution/Kurth6D.H:24]: Feel free to also add a paper/textbook reference here.

**Review by ax3l — COMMENTED** [review]
Awesome, thanks a lot!

Do you like to add a simple inputs set that uses this distribution, to make sure CI can run it? :)

File: src/ImpactX.cpp (hunk 1 above)
Comment (cemitch99) [reply src/ImpactX.cpp:357]: OK, will do.

File: src/particles/distribution/Kurth6D.H (hunk 3 above)
Comment (cemitch99) [reply src/particles/distribution/Kurth6D.H:24]: Yes, will do.

File: src/particles/distribution/Kurth6D.H (hunk 2 above)
Comment (cemitch99) [reply src/particles/distribution/Kurth6D.H]: This comment was copied/pasted from the Waterbag documentation.  It is equally valid here.  I think "length" can be replaced by "m".

File: examples/kurth/README.rst [hunk 4]
```diff
@@ -0,0 +1,22 @@
+.. _examples-kurth:
+
+Kurth Distribution in a Constant Focusing Channel
+==================================================
```
Comment (ax3l) [inline examples/kurth/README.rst]: Just `.rst` being picky to render this:
```suggestion
Kurth Distribution in a Constant Focusing Channel
=================================================
```

File: examples/kurth/README.rst [hunk 5]
```diff
@@ -0,0 +1,22 @@
+.. _examples-kurth:
```
Comment (ax3l) [inline examples/kurth/README.rst:1]: Awesome, add `examples/kurth/README.rst` in the table of contents in `docs/source/usage/examples.rst` please to include the file when it gets rendered to HTML/LaTeX/EPUB :)

File: src/particles/distribution/Waterbag.H [hunk 6]
```diff
@@ -24,7 +24,7 @@ namespace distribution
          * Return sampling from a 6D Waterbag distribution.
          *
          * @param sigx,sigy,sigt for zero correlation, these are the related
-         *                                   RMS sizes (length); ...
+         *                                   RMS sizes (in meters);
```
Comment (ax3l) [inline src/particles/distribution/Waterbag.H]:
```suggestion
         *                                   RMS sizes (in meters)
```

File: examples/kurth/README.rst (hunk 5 above)
Comment (ax3l) [reply examples/kurth/README.rst:1]: done via 13b09014294ffe1ca32ee8aab506369831bb0ed2

**Review by ax3l — APPROVED** [review]
Thank you :rocket: :sparkles:

**Pull request merged**